extension/injections/extensible_injector.cpp

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright (c) 2012-2017 Kanstantsin Chernik
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+//<-
+#include <cassert>
+#include <memory>
+//->
+#include <boost/di.hpp>
+
+namespace di = boost::di;
+
+template <class TDep>
+class dependency_proxy
```

Comment:
Did you consider inheriting from core::dependency directly?

Reply:
Yes, I tried but there is a problem with constructors of scope, as I remember instance scope

Reply:
'boost::di::v1_0_2::scopes::instance::scope<TExpected,TGiven,int>': no appropriate default constructor available	extension.injections_extensible_injector